src/node-backend/controller/homework-controller.js

```diff
@@ -11,19 +11,25 @@ import {
 import * as dbAdapter from "../adapters/database-adapter.js";
 import {checkForAchievement} from "../adapters/database-adapter.js";
 
+const queryResult= " SELECT COUNT( taskStatus) AS type FROM tasks, taskstatus WHERE assignedToStudent='student2' AND (taskStatus = 'done' OR taskStatus = 'finished' OR taskStatus = 'gradings')";
+
 export async function checkAchievements(username) {
     // check for achievements in database
+   try{
+        
     const queryResult = await dbAdapter.checkForAchievement(username);
```

Comment:
look at the function `checkForAchievement` in the database-adapter file, your query string and the `pool.query` call should be in there :) otherwise this is correct